ads/inabox/util.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * Executes a "fake vsync" read/write.
+ * This function exists mainly since the vsync service is not available for the
+ * inabox host script.
+ * It also helps with maintainability. Since the passed tasks have to define
+ * measure and mutate callbacks, it makes it harder for the calling code to be
+ * changed in a way that screws up the read-write order.
+ * Please note that this is NOT real vsync. Concurrent reads and writes ARE NOT
+ * BATCHED. This means that using this can still cause layout thrashing if it's
+ * being called more than once within the same frame. Use with caution.
+ * @param {!Window} win
+ * @param {{measure: (Function|undefined), mutate: !Function}} task
+ * @param {!Object=} opt_state
+ * @visibleForTesting
+ */
+// TODO(alanorozco): Figure out a longer-term solution
+export function fakeVsync(win, task, opt_state) {
```

Comment:
call it something else. `fake` is not a good word in production code
something like `runOnAnimationFrame` ?